### PR TITLE
Adding some more information to the readme, #45

### LINKED ISSUE #45: Add more details to SSH tunnel example
The README has this part about SSH tunneling:

> Both Matomo instances may be on different servers with proper firewall rules that restrict database access on target instance. In such case, the easiest way for source server to access target database is to create a ssh tunnel on new port (e.g. 3307) in another terminal. Then, execute to the above command with `--target-db-port=3307` instead to access port 3306 on target host. Example:
> 
> `ssh -NL 3307:localhost:3306 targetuser@targethost`

It would be great with more details, like where is the command supposed to be run? Also, what is the value of `targetuser@targethost` supposed to be set to?

Thanks!!

### PR DESCRIPTION
**Pull request by snake14**
### Description:

A customer recommended adding more information about SSH tunnelling and about an alternative.
Fixes: #45 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by AltamashShaikh — APPROVED** [review]
@snake14 The changes are fine.

I think we should ask someone from the support team if they want to add more info here

**Comment by AltamashShaikh** [comment]
@snake14 Also we should make this bold to make it more obvious.
Screenshot from 2022-11-14 09-59-20

**Comment by gitressa** [comment]
### New SSH tunneling text
Thanks! But It is still not clear to me where to run `ssh -NL 3307:localhost:3306 targetuser@targethost` ... my machine, the source or target? Perhaps the text could start with outlining this?

**Now**

> `ssh -NL 3307:localhost:3306 targetuser@targethost`
> 
>  This essentially maps port 3307 [...]

**Better?**

> `ssh -NL 3307:localhost:3306 targetuser@targethost`
> 
> Run the above command on THERIGHTPLACE. This essentially maps port 3307 [...]

Replace THERIGHTPLACE ^^ with the either "your own machine", "the source server" or "the target server".

### About `siteid`
In my opinion, it would be better to expand the text, and spell it out, so that it is really clear what will happen. Also, perhaps move it up the page "To start a migration execute ..."?:

> The migration tool will create a new website in the target Matomo, using the next available `siteid`, and copy all the data from the source website to this newly created target website. Remember to update the tracking code to match the new `siteid`.
> 
> To start a migration execute [...]

**Comment by gitressa** [comment]
### Migrating on target server

> An alternative to using an SSH tunnel is to make a backup of your MySQL database, copy it to the new server, import it into the database, and then migrate using that database name.

Maybe emphasize that you should probably import it into a temporary, separate database?

> An alternative to using an SSH tunnel is to make a backup of your MySQL database, copy it to the new server, import it into a temporary, separate database, and then migrate using that database name. Remember to delete the temporary database after completing the migration, and checking that everything works.

**Comment by snake14** [comment]
Thank you for the feedback @gitressa . It's no wonder you were still confused, because I was apparently confused and wrote the new text backwards. I have corrected that and tried to incorporate your suggestions.

File: README.md [hunk 1]
```diff
@@ -43,14 +46,19 @@ Then, execute to the above command with `--target-db-port=3307` instead to acces
 ```
 ssh -NL 3307:localhost:3306 targetuser@targethost
 ```
+This command should be run on the source Matomo server. It essentially maps port 3307 of the server to port 3306 of the server you are migrating to and is specified by `targethost`.
+The `targethost` should be replaced by a valid IP or domain name referencing the server you are migrating to and must be accessible to the server that you are migrating from.
+The `targetuser` should be replaced with a valid SSH user account setup on that server that you are migrating to.
+
+An alternative to using an SSH tunnel is to make a backup of your MySQL database, copy it to the new server, import it into a temporary database, and then migrate using that database name.
+Remember to delete the temporary database after completing the migration, and checking that everything works.
+More information about this process can be found in a [FAQ about moving between servers](https://matomo.org/faq/how-to-install/faq_76/).
```
Comment (AltamashShaikh) [inline README.md]:
```suggestion
More information about this process can be found [here](https://matomo.org/faq/how-to-install/faq_76/).
```

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 Just 1 suggestion apart from that all looks good.

File: README.md (hunk 1 above)
Comment (gitressa) [reply README.md]: I dislike "here" links, and would keep the original version.

If anything, the link text could be the original title: "How can I move Matomo from one server to another, also migrating the data from one mysql server to another?"

**Comment by gitressa** [comment]
Thanks @snake14, the text is clearer now, and makes sense. I had prepared a longer text about the reversals, but thought I'd start here and see if you caught them, which you did :-)  

However, one sentence still puzzles me:

> The `targethost` should be replaced by a valid IP or domain name referencing the server you are migrating to and must be accessible to the server that you are migrating from.

How do you make the `targethost` accessible to the server that you are migrating from? Do you need to create a SSH user on source server and upload SSH public keys to target server? If that's the case, it should probably be mentioned.

Also, has SSH tunneling been verified to work by one of you guys?

File: README.md (hunk 1 above)
Comment (AltamashShaikh) [reply README.md]: Even the FAQ title looks good 👍🏻

Comment (snake14) [reply README.md]: Yeah. I'm pretty sure that I read in one of our wiki pages that we should use more descriptive text than just `here` or `FAQ` when linking to things in our documentation. Using the full name of the FAQ will work :+1:

**Comment by snake14** [comment]
Thank you for your graciousness @gitressa . I have added a little more detail and some more links for reference.

I haven't tried SSH tunnelling with the Migration plugin, but I have used it in the past. It's common to use it to map a local port to a remote database.

**Comment by gitressa** [comment]
Thanks @snake14, the latest version with the added details about SSH requirements makes sense, answering the most relevant questions. So that part is done.

Ideally (for the perfect review) someone would go through the steps in the guide, and be able to complete a migration between two firewalled servers, to verify that the steps outlined give enough info to complete the task.

Either by someone with very basic SSH knowledge, or someone who "adapts" the tech knowledge level mentally, and tries to think as such a user (if it makes sense). But I also realize that this takes more time and resources, and maybe not realistic ...

**Comment by snake14** [comment]
Thank you @gitressa . Yeah. We usually try to do functional testing as part of our code reviews, but that's not really applicable for a readme change. I can see if a member of our support team can take a look at the changes and see if they have any feedback.

**Comment by gitressa** [comment]
Sounds good. Thanks for being open to updating the documentation, and being so patient with all my suggestions.

**Pull request merged**